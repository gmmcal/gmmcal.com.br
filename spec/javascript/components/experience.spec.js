import { mount } from '@vue/test-utils'
import VueMarkdown from '@/components/markdown'
import Experience from '@/components/experience'
import create from '../settings/utils'

describe('Experience', () => {
  let props, wrapper

  beforeEach(() => {
    props = {
      propsData: {
        experience: create('experience')
      }
    }

    wrapper = mount(Experience, props)
  })

  it('is a Vue instance', () => {
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  it('has data properties', () => {
    const expected = ['I18n']
    const received = Object.keys(wrapper.vm.$data)
    expect(received).toEqual(expected)
  })

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  describe('Content', () => {
    it('has dates', () => {
      const el = wrapper.find('.year')
      expect(el.text()).toBe('Jan / 2015Jan / 2016')
    })

    describe('Contains company details', () => {
      it('with company name', () => {
        const el = wrapper.findAll('h4 span').at(0)
        expect(el.text()).toBe('Amazing company')
      })

      it('with city', () => {
        const el = wrapper.findAll('h4 span').at(2)
        expect(el.text()).toBe('Great city')
      })

      it('with country', () => {
        const el = wrapper.findAll('h4 span').at(3)
        expect(el.text()).toBe('Some country')
      })
    })

    it('has a course', () => {
      const el = wrapper.find('h5')
      expect(el.text()).toBe('Ninja developer')
    })

    it('has a description', () => {
      const el = wrapper.find('.description')
      expect(el.text()).toBe('Lorem ipsum nulus dolor')
    })

    describe('without end date', () => {
      beforeEach(() => {
        props = {
          propsData: {
            experience: create('experience', { end_date: null })
          }
        }

        wrapper = mount(Experience, props)
      })

      it('has Currently text', () => {
        const el = wrapper.find('.year')
        expect(el.text()).toBe('Jan / 2015Currently')
      })
    })
  })

  describe('Child components', () => {
    it('includes VueMarkdown', () => {
      const child_component = wrapper.find(VueMarkdown)
      expect(child_component.isVueInstance()).toBeTruthy()
    })
  })
})
