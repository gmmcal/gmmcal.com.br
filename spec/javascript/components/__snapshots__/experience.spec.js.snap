// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Experience renders correctly 1`] = `
<div class="row work-details">
  <div class="col-xs-12 col-sm-12 col-md-3 col-lg-2">
    <div class="year">Jan / 2015<br>Jan / 2016</div>
  </div>
  <div class="col-xs-12 col-sm-12 col-md-9 col-lg-10 right-area">
    <div class="arrowpart"></div>
    <div class="ex-con">
      <h4><span>Amazing company</span> - <span><span>Great city</span> - <span>Some country</span></span></h4>
      <h5>Ninja developer</h5>
      <div class="description">
        <p>Lorem ipsum nulus dolor</p>
      </div>
    </div>
  </div>
</div>
`;
